javalib/src/main/scala/java/util/Date.scala

```diff
@@ -36,82 +38,91 @@ class Date private (private val date: js.Date) extends Object
   def this(year: Int, month: Int, date: Int) =
     this(year, month, date, 0, 0, 0)
 
-  def this(date: Long) = this(new js.Date(date.toDouble))
-
   @Deprecated
-  def this(date: String) = {
-    this({
-      val jsDate = new js.Date(date)
-      if (java.lang.Double.isNaN(jsDate.getTime()))
-        throw new IllegalArgumentException
-      jsDate
-    })
-  }
+  def this(date: String) = this(Date.parse(date))
 
-  def after(when: Date): Boolean = date.getTime() > when.date.getTime()
+  def after(when: Date): Boolean = millis > when.millis
 
-  def before(when: Date): Boolean = date.getTime() < when.date.getTime()
+  def before(when: Date): Boolean = millis < when.millis
 
-  override def clone(): Object = new Date(new js.Date(date.getTime()))
+  override def clone(): Object = new Date(millis)
 
   override def compareTo(anotherDate: Date): Int =
-    java.lang.Double.compare(date.getTime(), anotherDate.date.getTime())
+    java.lang.Long.compare(millis, anotherDate.millis)
 
   override def equals(obj: Any): Boolean = obj match {
-    case d: Date => d.date.getTime() == date.getTime()
+    case d: Date => d.millis == millis
     case _       => false
   }
 
-  override def hashCode(): Int = date.getTime().hashCode()
+  override def hashCode(): Int = millis.hashCode()
 
   @Deprecated
-  def getDate(): Int = date.getDate().toInt
+  private def asDate(): js.Date = {
+    if (!isSafeJSDate()) {
+      throw new IllegalArgumentException(
+          f"cannot convert this java.util.Date ($millis millis) to a js.Date")
```

Comment:
```suggestion
          s"cannot convert this java.util.Date ($millis millis) to a js.Date")
```
There's no need to depend on `java.util.Formatter` for this.